kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkTask.java

```diff
@@ -309,12 +309,24 @@ public Map<TopicPartition, OffsetAndMetadata> preCommit(
   ) {
     Map<TopicPartition, OffsetAndMetadata> offsetsToCommit = new HashMap<>();
 
+    /* Most of the time, `offsets` contains all `topicPartitionWriters` (on regular flush intervals & when closing the connector).
+     * However, in case of a revoked partition/s, `offsets` will contain only topic partitions that are going to be revoked.
```

Comment:
How do we know that `offsets` are the ones of the partitions that need to be revoked?

Reply:
I went over confluence's code [here](https://github.com/confluentinc/kafka/blob/d68be3dd4aa01d3504e720744fa40f9645578837/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java#L430)